casper/src/main/scala/coop/rchain/casper/EstimatorHelper.scala

```diff
@@ -21,14 +21,14 @@ object EstimatorHelper {
   def chooseNonConflicting[F[_]: Monad: Log: BlockStore](
       blockHashes: Seq[BlockHash],
       dag: BlockDagRepresentation[F]
-  ): F[Seq[BlockMessage]] = {
-    def nonConflicting(b: BlockMessage): BlockMessage => F[Boolean] =
+  ): F[Seq[BlockMetadata]] = {
+    def nonConflicting(b: BlockMetadata): BlockMetadata => F[Boolean] =
       conflicts[F](_, b, dag).map(b => !b)
 
     for {
-      blocks <- blockHashes.toList.traverse(hash => ProtoUtil.unsafeGetBlock[F](hash))
+      blocks <- blockHashes.toList.traverse(hash => ProtoUtil.unsafeGetBlockMetadata[F](hash, dag))
```

Comment:
why are ` ProtoUtil.unsafeGetBlockMetadata][F]` and ` ProtoUtil.unsafeGetBlock[F]` unsafe?

Reply:
They are unsafe as in "will crash if the requested block is not actually in the block store/dag storage".

Reply:
then they should be just `getBlock` or `tryGetBlock` if we need anything more to tell that than just the type. `unsafe` should be reserved for dirty stuff like impure effects underneath, or throwing exceptions which those methods should not be doing.

Reply:
I renamed `unsafeGetBlockMetadata` added in this PR to `getBlockMetadata` and changed its signatured from `F[_]: Monad` to `F[_]: Sync`. Doing the same with `unsafeGetBlock` (and other "unsafe" methods in ProtoUtil) would make this relatively focused PR to touch the entire casper code base.

Reply:
Can you do that in a follow-up PR then?

Reply:
Sure